influxql/parser.go

```diff
@@ -510,7 +510,7 @@ func (p *Parser) parseUInt64() (uint64, error) {
 // This function assumes the DURATION token has already been consumed.
 func (p *Parser) parseDuration() (time.Duration, error) {
 	tok, pos, lit := p.scanIgnoreWhitespace()
-	if tok != DURATION_VAL && tok != INF {
+	if tok != DURATION_VAL && tok != INF && tok != NUMBER {
```

Comment:
`FormatDuration` [does not print units if it has to print the duration in microseconds](https://github.com/influxdb/influxdb/blob/fix-statement-strings/influxql/parser.go#L2421-L2438).

When that happens, if we parse the statement again, the token parses as a NUMBER instead of a DURATION_VAL.

Now that you've pointed it out and now that I've read the EBNF closer, I see that I should not have changed `parseDuration` but I should have changed `FormatDuration` to include units because a `duration_lit` *must* have units. Not sure why I thought the unit was optional on a `duration_lit` – good catch. I'll remove the commit that includes this line, and I'll add a new commit that ensures FormatDuration prints units in the case of microseconds.